catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -244,6 +245,19 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
         responseProps.put(SUGGESTION_RESULT_KEY, (Serializable) suggestionResults);
       }
 
+      if (solrResponse.getSpellCheckResponse() != null
+          && solrResponse.getResults().size() == 0
+          && solrResponse.getSpellCheckResponse().getCollatedResults().size() != 0) {
+        query.set("q", findQueryToResend(query, solrResponse));
+        solrResponse = client.query(query, METHOD.POST);
```

Comment:
❓ Are we resending the query? If so, why can't we attach the spellcheck options onto the original query?

Reply:
I'm not sure if I understand. Are you saying to apply the spellcheck on the query before we actually perform the query? What about the case where you have misspelled data in the databased, and want to search for those?

Reply:
Gotcha I misunderstood what this was doing.